src/query-client/hooks/item.js

```diff
@@ -0,0 +1,97 @@
+import { useQuery } from 'react-query';
+import { List, Map } from 'immutable';
+import * as Api from '../api';
+import {
+  ALL_ITEMS_KEY,
+  buildItemChildrenKey,
+  buildItemKey,
+  buildItemMembersKey,
+  buildItemParentsKey,
+  buildMemberKey,
+} from '../config/keys';
+
+export default (queryClient, queryConfig) => {
+  const { retry, cacheTime, staleTime } = queryConfig;
+  const defaultOptions = {
+    retry,
+    cacheTime,
+    staleTime,
+  };
+
+  const useItem = (id) =>
+    useQuery({
+      queryKey: buildItemKey(id),
+      queryFn: () => Api.getItem({ id }, queryConfig).then((data) => Map(data)),
+      enabled: Boolean(id),
+      onSuccess: async (item) => {
+        // save items in their own key
+        // eslint-disable-next-line no-unused-expressions
+        queryClient.setQueryData(buildItemKey(id), Map(item));
+      },
+      ...defaultOptions,
+    });
+
+  const useAllItems = () =>
+    useQuery({
+      queryKey: ALL_ITEMS_KEY,
+      queryFn: () => Api.getAllItems(queryConfig).then((data) => List(data)),
+      onSuccess: async (items) => {
+        // save items in their own key
+        // eslint-disable-next-line no-unused-expressions
+        items?.forEach(async (item) => {
```

Comment:
I might be wrong, but do you need this to be async? Same for `onSuccess` in many other hooks.